### ISSUE TITLE
Bug: readonly arrays not allowed as covariance matrix for new gvars

### ISSUE THREAD

**Issue by Gattocrucco** [gplepage/gvar#27]
The factory does not allow me to make new gvars from a read-only covariance matrix when I use the `fast=True` option:

```python
import gvar
import numpy as np

mean = np.zeros(1)
cov = np.ones((1, 1))
cov.flags['WRITEABLE'] = False
g = gvar.gvar(mean, cov, fast=True)
```

Output:

```python
File _gvarcore.pyx:938, in gvar._gvarcore.GVarFactory.__call__()

File _svec_smat.pyx:354, in gvar._svec_smat.smat.append_diag_m()

File stringsource:658, in View.MemoryView.memoryview_cwrapper()

File stringsource:349, in View.MemoryView.memoryview.__cinit__()

ValueError: buffer source array is read-only
```

This is weird because I do not expect `gvar.gvar` to write to my matrix, and probably indeed it's not writing anything. I guess it could be a Cython typing complaint.

**Comment by gplepage** [comment]
Clearly a bug. I will fix it in the next release.

**Comment by gplepage** [comment]
I will fix this too in the next release. I need to add a few ``const``s to the Cython code.

Note, by the way, that ``svec`` is not really part of the public interface to ``gvar``, which is why it isn't mentioned in the documentation. The same is true of ``smat`` although I couldn't keep it entirely out of the documentation. This is because over the years I have been playing with various alternatives to ``svec`` and ``smat`` --- they could both disappear from a future release.

**Comment by Gattocrucco** [comment]
Ok

**Comment by gplepage** [comment]
Fixed in version 11.10.1.
